src/core/operations/MultiTapCipherDecode.mjs

```diff
@@ -0,0 +1,58 @@
+/**
+ * @author Necron3574 [kaustubhbm3574@gmail.com]
+ * @copyright Crown Copyright 2021
+ * @license Apache-2.0
+ */
+
+import Utils from "../Utils.mjs";
+import Operation from "../Operation.mjs";
+import OperationError from "../errors/OperationError.mjs";
+
+/**
+ * Multi Tap Cipher Decode operation
+ */
+class MultiTapCipherDecode extends Operation {
+
+    /**
+     * MultiTapCipherDecode constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "Multi Tap Cipher Decode";
+        this.module = "Default";
+        this.description = "It is a simple substitution cipher which substitutes the alphabets to their corresponding taps on a mobile phone.<br><br>e.g. <code>adg</code> becomes <code>2-3-4</code>";
+        this.infoURL = "https://wikipedia.org/wiki/Multi-tap";
+        this.inputType = "string";
+        this.outputType = "string";
+        this.args = [
+            {
+                name: "Delimiter",
+                type: "option",
+                value: ["Hyphen","Space", "Comma", "Semi-colon", "Colon", "Line feed"]
+            }
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+        if(input.length == 0)
+          return "";
+        const delim = Utils.charRep(args[0] || "Hyphen");
+        var emap = {"2":"A","22":"B","222":"C","3":"D","33":"E","333":"F","4":"G","44":"H","444":"I","5":"J","55":"K","555":"L","6":"M","66":"N","666":"O","7":"P","77":"Q","777":"R","7777":"S","8":"T","88":"U","888":"V","9":"W","99":"X","999":"Y","9999":"Z"," ":" "}
```

Comment:
Is it the case that the space between words is always represented by a space, or is a different space delimiter ever used? It might be good to store this dict in a separate file so that both the encoding and decoding can use the same one, to reduce duplication.

Reply:
Thanks for the feedback,
Since this is the decode operation, the input should be of the form `222#3#33#333` where the hash represents a delimiter chosen by the user which is already a configurable option.
Also regarding the dictionaries, the way they are saved is unidirectional so I can only refer to the value using the key. So I'm not sure if both encoding and decoding can use the same one.  I can fit in both the encryption and decryption values into the same dictionary so that both the encoding and decoding can use the same one but I feel this maybe more readable. However I can save these dictionaries to another file or do the above if you want me to.

Reply:
I was wondering if the spaces between words will always be indicated by " "? It is possible the user could have got the input from another source, and so if there are any other conventions (for example using "-" to mean space) it would be good to be adaptable to those too. However, I haven't seen this problem in real use before so I don't know whether these other conventions exist 🙂

I was thinking that if they were in a separate place you could then store the mapping in one direction, and if the other direction was needed reverse the dictionary. There are some examples about how you could do that [here.](https://stackoverflow.com/questions/23013573/swap-key-with-value-json) I think that this would be preferable if possible so that the mapping is only stored in one place.

Reply:
Yep. I updated the above changes.